src/ott/tools/soft_sort.py

```diff
@@ -450,6 +451,84 @@ def _quantile(
   return apply_on_axis(_quantile, inputs, axis, q, weight, **kwargs)
 
 
+def multiv_cdf_quantile_maps(
+    inputs: jnp.ndarray,
+    targets: Optional[jnp.ndarray] = None,
+    target_sampler: Optional[Callable[[jax.random.PRNGKey, Tuple[int, int]],
+                                      jnp.ndarray]] = None,
+    key: Optional[jax.random.PRNGKey] = None,
+    target_num_samples: Optional[int] = None,
+    cost_fn: Optional[jnp.ndarray] = None,
+    epsilon: Optional[float] = None,
+    input_weights: Optional[jnp.ndarray] = None,
+    target_weights: Optional[jnp.ndarray] = None,
+    **kwargs: Any
+) -> [
+    Callable[[jnp.ndarray], jnp.ndarray], Callable[[jnp.ndarray], jnp.ndarray]
+]:
+  r"""Returns multivariate CDF and quantile maps, given input samples.
+
+  Implements the multivariate generalizations for CDF and quantiles proposed in
+  :cite:`chernozhukov:17`. The reference measure is assumed to be the uniform
+  measure by default, but can be modified. For consistency, the reference
+  measure should be symmetrically centered around
+  :math:`(\tfrac{1}{2},\cdots,\tfrac{1}{2})` and suppported on :math:`[0,1]^d`.
+
+  The implementation return two entropic map estimators, one for the CDF map,
+  the other for the quantiles map.
+
+  Args:
+    inputs: 2D array of `[n, d]` vectors.
+    targets: 2D array of `[m,d]` vectors. If not passed explicitly, the random
+      sampler specified in `target_sampled` is used to create these samples from
+      reference measure.
+    target_sampler: When ``targets`` is not passed, Callable that takes a
+      ``key`` and shape specified by ``target_num_samples`` to sample a target
+      points.
+    key: `rng` key used in ``target_sampler``
+    target_num_samples: number of points generated in the target distribution.
+    cost_fn: cost function used to compare ``inputs`` and ``targets``. Passed on
+      to instantiate a {class}`~ott.geometry.pointcloud.PointCloud` object. This
```

Comment:
Use :class:\`~ott.geometry.pointcloud.PointCloud\` (and similarly in other places).